rules/prefer-number-is-integer.js

```diff
@@ -0,0 +1,158 @@
+'use strict';
+
+const {methodCallSelector} = require('./selectors/index.js');
+const isSameReference = require('./utils/is-same-reference.js');
+
+const MESSAGE_ID = 'preferNumberIsInteger';
+const MESSAGE_ID_SUGGEST = 'preferNumberIsIntegerSuggestion';
+const messages = {
+	[MESSAGE_ID]: 'Replace `{{original}}` with `Number.isInteger({{variable}})`.',
+	[MESSAGE_ID_SUGGEST]: 'Prefer `Number.isInteger()` for integer checks.',
+};
+
+const equalsSelector = ':matches([operator="==="],[operator="=="])';
+
+// `value % 1 === 0`
+const modulo1Selector = [
+	'BinaryExpression',
+	'[left.type="BinaryExpression"]',
+	'[left.operator="%"]',
+	'[left.right.value=1]',
+	equalsSelector,
+	'[right.value="0"]',
+].join('');
+
+// `(value ^ 0) === value`
+// `(value | 0) === value`
+const mathOperatorSelector = [
+	'BinaryExpression',
+	'[left.type="BinaryExpression"]',
+	`:matches(${['^', '|'].map(operator => `[left.operator="${operator}"]`).join(',')})`,
+	'[left.right.value=0]',
+	equalsSelector,
+].join('');
+
+// `Number.parseInt(value,10) === value`
+const numberParseIntSelector = [
+	'BinaryExpression',
+	'[left.type="CallExpression"]',
+	'[left.callee.type="MemberExpression"]',
+	'[left.callee.object.name="Number"]',
+	'[left.callee.property.name="parseInt"]',
+	'[left.arguments.1.value=10]',
+	equalsSelector,
+].join('');
+
+// `_.isInteger(value)`
+const lodashIsIntegerSelector = [
+	methodCallSelector({method: 'isInteger', objects: ['_', 'lodash', 'underscore']}),
+].join('');
+
+// `Math.round(value) === value`
+const mathRoundSelector = [
+	'BinaryExpression',
+	methodCallSelector({method: 'round', object: 'Math', path: 'left'}),
+	equalsSelector,
+].join('');
+
+// `~~value === value`
+const bitwiseNotSelector = [
+	'BinaryExpression',
+	'[left.type="UnaryExpression"]',
+	'[left.operator="~"]',
+	'[left.argument.type="UnaryExpression"]',
+	'[left.argument.operator="~"]',
+	equalsSelector,
+].join('');
+
+function createNodeListener(sourceCode, variableGetter) {
+	return node => {
+		const variable = variableGetter(node);
+
+		if (!variable) {
```

Comment:
We should not limit this to  `Identifier` and `MemberExpression`, should allow anything

```js
_.isInteger(a ? b : c)
```

Reply:
// @tristanHessell